audits/token/202312-threat-model-preset-erc1155.md

```diff
@@ -31,12 +44,79 @@ Permit is added to allow for Gasless transactions from the token owners.
 - Override `uri` to return `baseURI` field to keep in standard with ImmutableERC721
 - Added `baseURI` to replace `uri` to encourage the usage of `baseURI`
 
-
 ## Attack Surfaces
 
-ERC1155 only has `setApproveForAll` as it's approval method. Meaning any flow that requires a 3rd party to operator on a set of tokens owned by another wallet will grant the third party access to all of that specific wallet's tokens. The third party needs to be entirely trustworthy. The owner needs to be diligent on revoking unrestricted access when not needed. 
-
-We can consider implementing a more complicated approval schema if needed. i.e by token id or by token id and amount. 
+ERC1155 only has `setApprovalForAll` as it's approval method. Meaning any flow that requires a 3rd party to operate on a set of tokens owned by another wallet will grant the third party access to all of that specific wallet's tokens. The third party needs to be entirely trustworthy. The owner needs to be diligent on revoking unrestricted access when not needed.
+
+The contract has no access to any funds. Additional risks can come from compromised keys that are responsible for managing the admin roles that control the collection. As well as permits and approves if an user was tricked into creating a permit that can be validated by a malicious eip1271 wallet giving them permissions to the user's token.
+
+Potential Attacks:
+
+- Compromised Admin Keys:
+  - The compromised keys are able to assign the `MINTER_ROLE` to malicious parties and allow them to mint tokens to themselves without restriction
+- Compromised Offchain auth:
+  - Since EIP4494 combined with EIP1271 relies on off chain signatures that are not standard to the ethereum signature scheme, user auth info can be compromised and be used to create valid EIP1271 signatures.
+
+## Attack Mitigation
+
+- The contract contains access control patterns to limit the vulnerabilty of compromised roles
+- The keys associated with the `DEFAULT_ADMIN_ROLE` should be operated by secure measures, for example a multi-signature wallet such that an attacker would need to compromise multiple signers simultaneously, or a securely stored hardware wallet.
+
+### Externally Visible Functions
+
+An attacker could formulate an attack in which they send one or more transactions that execute one or more of these functions.
+
+Functions that _change_ state:
+| Name | Function Selector | Access Control |
+| ------------------------------------------------------------- | ----------------- | --------------------- |
+| burn(address,uint256,uint256) | f5298aca | None - permisionless |
```

Comment:
I would expect the access control is that only the token owner can call burn.

Reply:
thanks, have addressed all state changing functions that previously mentioned being permissionless